.github/workflows/smart_contracts.yaml

```diff
@@ -32,24 +37,51 @@ env:
   NET_NAME: "main"
 
 jobs:
+  preparation:
+    runs-on: "ubuntu-latest"
+    steps:
+      - if: "github.event_name == 'workflow_dispatch'"
+        shell: "sh"
+        run: 'echo "# Run parameters:" >> "${GITHUB_STEP_SUMMARY}"'
+      - if: "github.event_name == 'workflow_dispatch' && fromJSON(github.event.inputs.skip_checks)"
+        shell: "sh"
+        run: 'echo "## Skip checks" >> "${GITHUB_STEP_SUMMARY}"'
+      - if: "github.event_name == 'workflow_dispatch' && fromJSON(github.event.inputs.optimized_build)"
+        shell: "sh"
+        run: 'echo "## Building optimized binaries" >> "${GITHUB_STEP_SUMMARY}"'
   check_formatting:
     runs-on: "ubuntu-latest"
+    needs:
+      - "preparation"
     steps:
       - uses: "actions/checkout@v3"
+        if: "github.event_name != 'workflow_dispatch' || !fromJSON(github.event.inputs.skip_checks)"
```

Comment:
a lot of repetition!!! please figure out a solution to avoid it! In addition, why it's part of this PR?